src/Browscap/Command/CheckDuplicateTestsCommand.php

```diff
@@ -0,0 +1,82 @@
+<?php
+declare(strict_types = 1);
+namespace Browscap\Command;
+
+use Browscap\Helper\LoggerHelper;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CheckDuplicateTestsCommand extends Command
+{
+    protected function configure() : void
+    {
+        $this
+            ->setName('check-duplicate-tests')
+            ->setDescription('checks the test cases for duplicates');
+    }
+
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return int|null null or 0 if everything went fine, or an error code
+     */
+    protected function execute(InputInterface $input, OutputInterface $output) : ?int
+    {
+        $loggerHelper = new LoggerHelper();
+        $logger       = $loggerHelper->create($output);
+
+        $data = [];
+
+        $checks          = [];
+        $sourceDirectory = __DIR__ . '/../../../tests/issues/';
+        $iterator        = new \RecursiveDirectoryIterator($sourceDirectory);
+
+        $errors = false;
+
+        foreach (new \RecursiveIteratorIterator($iterator) as $file) {
+            /** @var $file \SplFileInfo */
+            if (!$file->isFile() || 'php' !== $file->getExtension()) {
+                continue;
+            }
+
+            $tests = require $file->getPathname();
+
+            foreach ($tests as $key => $test) {
+                if (isset($data[$key])) {
+                    $logger->error('Test data is duplicated for key "' . $key . '"');
```

Comment:
Isn't this already done in the test classes themselves (or maybe just the Full test)?

https://github.com/browscap/browscap/blob/master/tests/UserAgentsTest/FullTest.php#L154

Why are you moving this to a separate command?

Possible to re-use the code in the `FullTest.php` file and here?

Reply:
In the testfiles this is done after the ini files are built. I added this to test this before the build.

Reply:
I moved the duplicate code to a helper class